Feature/session handling integration

aburi-ost opened the pull request:
Hi Sven

In diesem PR habe ich unter anderem die struktur der NoteEntries um creationDate erweitert. Kann also sein dass Du deine Default-Werte für die Datenbank neu erstellen musst. Hier falls du gleich was copy-pasten möchtest:

{"creationDate":{"$$date":1689087718156},"dueDate":"0001-02-04","title":"ccc","importance":"3","state":"OPEN","description":"cCc","_id":"U24uKqD37SD8H7E3"}
{"creationDate":{"$$date":1689087267824},"dueDate":"0001-01-01","title":"bbbb","importance":"4","state":"OPEN","description":"bbbbbb","_id":"bWRMEGy8apiUCIra"}
{"creationDate":{"$$date":1689087251974},"dueDate":"0003-02-01","title":"aaa","importance":"5","state":"OPEN","description":"aaaaa","_id":"w4vX1YSewrps6TlV"}


Zeit für eine Pause :)

aburi-ost committed:
Remove update route
Add method-override middleware
Adjust request handling to Gfeller methodology

aburi-ost committed:
Add dark mode to index and update

aburi-ost committed:
Remove unneeded comments

aburi-ost committed:
Add session-middleware handling to existing order and filter functions
Add order functionality in descending direction

aburi-ost committed:
Add creationDate to noteEntry
Add sort functionality for creationDate

aburi-ost committed:
Rename method-override.js to method-override.mjs

aburi-ost committed:
Replace string comparison with const member comparison for OrderBy and OrderDirection

aburi-ost committed:
Replace passing of individual sort settings to getAll() with passing the userSettings

soolerseven commented on controller/details-controller.mjs:
```diff
@@ -1,29 +1,57 @@
-        res.render('details')
+
+    redirect = (req, res, id) => {
+        if ('update_overview_button' in req.body || 'create_overview_button' in req.body) {
```
Von Name des Buttons auf den Wert ändern

aburi-ost replied:
Mir ist noch aufgefallen, dass im Body jeweils nur der Button mitgeschickt wird, welcher betätigt wird.
Kann das zu Problemen führen, falls der code z.B. req.body.update_button zugreift, dieser jedoch im body gar nicht vorhanden ist? Oder wandelt das Javascript einfach in ein "false" um?

soolerseven commented on routes/details-routes.mjs:
```diff
@@ -1,9 +1,10 @@
+detailsRouter.get('/', detailsController.detailsEmpty)
 detailsRouter.post('/', detailsController.createEntry)
+detailsRouter.get("/:id/", detailsController.detailsByID);
+detailsRouter.put('/:id/', detailsController.updateEntry)
```
Evtl. Patch?

aburi-ost replied:
Basierend auf folgender Definition denke ich, dass put angebrachter ist:
"PUT modifies a record's information and creates a new record if one is not available, and PATCH updates a resource without sending the entire body in the request."

soolerseven commented on services/noteEntry-store.mjs:
```diff
@@ -1,63 +1,58 @@
+        })
+    }
+    orderByCreationDate = (DataBaseEntries, orderDirection) => {
+        DataBaseEntries.sort((a, b) => {
```
Vereinfachen: Direkt in orderLogic reinschreiben

soolerseven commented on utils/method-override.mjs:
```diff
@@ -0,0 +1,9 @@
+import override from "method-override"; // Importing the 'method-override' module
+
+export const overrideMiddleware = override(function (req, res) { // Creating the overrideMiddleware function using the 'method-override' module
```
ToDo: aufräumen

soolerseven reviewed (changes requested):
Wie zusammen besprochen

aburi-ost committed:
Change body->button access for redirect in details-controller.mjs
Clean up details-controller.mjs
Inline creationDate into function call in noteEntry-store.mjs
